src/main/java/pl/betoncraft/betonquest/events/TakeEvent.java

```diff
@@ -73,39 +68,41 @@ protected Void execute(final String playerID) throws QuestRuntimeException {
             for (final CheckType type : checkOrder) {
                 switch (type) {
                     case INVENTORY:
-                        checkInventory(player); break;
+                        checkInventory(player);
+                        break;
                     case ARMOR:
-                        checkArmor(player); break;
+                        checkArmor(player);
+                        break;
                     case BACKPACK:
-                        checkBackpack(playerID); break;
+                        checkBackpack(playerID);
+                        break;
                 }
             }
             notifyPlayer(playerID, questItem, deleteAmount);
         }
         return null;
     }
 
-    @SuppressFBWarnings("NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE")
     private void checkInventory(final Player player) {
-        final List<ItemStack> inv = new ArrayList<>(Arrays.asList(player.getInventory().getContents()));
-        final ItemStack[] newInv = removeDesiredAmount(player, inv);
-        player.getInventory().setContents(newInv);
+        final ItemStack[] inventory = player.getInventory().getStorageContents();
+        final ItemStack[] newInv = takeDesiredAmount(player, inventory);
+        player.getInventory().setStorageContents(newInv);
```

Comment:
Although this change from `getContents()` to `getStorageContents()` does prevent armor from being taken it excludes the off hand, so no items will be taken from there.